### PR TITLE
Add additional filter check for reports and regressions

### PR DESCRIPTION
**Pull request by nharper285**
## Summary of the Pull Request

_What is this about?_
Add additional filter condition to catch `source-coverage.json` files. 

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

File: src/ApiService/ApiService/onefuzzlib/Reports.cs [hunk 1]
```diff
@@ -28,9 +28,9 @@ public class Reports : IReports {
 
     public async Async.Task<IReport?> GetReportOrRegression(Container container, string fileName, bool expectReports = false, params string[] args) {
         var filePath = string.Join("/", new[] { container.String, fileName });
-        if (!fileName.EndsWith(".json", StringComparison.Ordinal)) {
+        if (!fileName.EndsWith(".json", StringComparison.Ordinal) || fileName.Contains("source-coverage", StringComparison.Ordinal)) {
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/Reports.cs:31]: I think `StringComparison.InvariantCultureIgnoreCase` is the recommended option: https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings#specifying-string-comparisons-explicitly

**Review by tevoinea — APPROVED** [review]

File: src/ApiService/ApiService/onefuzzlib/Reports.cs (hunk 1 above)
Comment (nharper285) [reply src/ApiService/ApiService/onefuzzlib/Reports.cs:31]: Ah good to know, thank you!

Comment (Porges) [reply src/ApiService/ApiService/onefuzzlib/Reports.cs]: For machine-readable stuff like this filename `OrdinalIgnoreCase` is more suited. The culture-sensitive versions can be _very_ weird if you're not expecting it:

```csharp
"source-coverage".Contains("C­", StringComparison.InvariantCultureIgnoreCase)
```

File: src/ApiService/ApiService/onefuzzlib/Reports.cs [hunk 2]
```diff
@@ -28,9 +28,9 @@ public class Reports : IReports {
 
     public async Async.Task<IReport?> GetReportOrRegression(Container container, string fileName, bool expectReports = false, params string[] args) {
         var filePath = string.Join("/", new[] { container.String, fileName });
-        if (!fileName.EndsWith(".json", StringComparison.Ordinal)) {
+        if (!fileName.EndsWith(".json", StringComparison.Ordinal) || fileName.Contains("source-coverage", StringComparison.InvariantCultureIgnoreCase)) {
```
Comment (Porges) [inline src/ApiService/ApiService/onefuzzlib/Reports.cs:31]: Should this not be:
```suggestion
        if (!fileName.EndsWith(".json", StringComparison.Ordinal) || !fileName.Contains("source-coverage", StringComparison.InvariantCultureIgnoreCase)) {
```

**Comment by Porges** [comment]
We should also be able to write a test with this using the existing implementations we have for Azurite stuff... let me know if you want any help putting one together.

File: src/ApiService/ApiService/onefuzzlib/Reports.cs (hunk 2 above)
Comment (nharper285) [reply src/ApiService/ApiService/onefuzzlib/Reports.cs:31]: @tevoinea can confirm, but it was my understanding that we don't want to parse the file if its name starts `source-config`.

Comment (tevoinea) [reply src/ApiService/ApiService/onefuzzlib/Reports.cs:31]: That's right, we don't want to parse the file is the name _ends with_ `source-coverage.json`

**Pull request merged**